### PR TITLE
Add human-readable duration parsing for MAGPIE_TIMEOUT

### LINKED ISSUE #28: Add configurable timeout and size limits with unit selection
## Summary

The CLI timeout is currently hard-coded to 30 minutes. This should be configurable via environment variable or config file, with support for human-readable units.

### Timeout configuration
- Environment variable: `MAGPIE_TIMEOUT`
- Support units: `30s`, `5m`, `1h`, etc.
- Default: `30m`

### Size limits (future)
- Environment variable: `MAGPIE_MAX_SIZE`
- Support units: `100MB`, `2GB`, etc.
- Default: TBD

## Implementation notes

- Add unit parsing utility (e.g., `parse_duration("30m")` -> 1800 seconds)
- Add to `MagpieSettings` in config.py
- Update CLI client to use configured timeout
- Document in user guide

---
Generated with Claude Code (Opus 4.5)

### PR DESCRIPTION
**Pull request by duplico**
## Summary
- Adds `parse_duration()` function to support human-readable formats: `30s`, `5m`, `1h`, `1h30m`, or plain seconds
- Updates `get_timeout()` to use duration parsing for the `MAGPIE_TIMEOUT` environment variable
- Adds comprehensive tests for duration parsing including edge cases and error handling

Closes #28

## Test plan
- [x] Unit tests pass for all duration formats: seconds (30s), minutes (5m), hours (1h), compound (1h30m)
- [x] Plain numbers still work as seconds
- [x] Invalid formats gracefully fall back to default timeout
- [x] Case-insensitive parsing (5M, 1H work)
- [x] Whitespace handling

---
Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/magpie/cli/config.py [hunk 1]
```diff
@@ -121,6 +122,67 @@ def get_token(
...
+_DURATION_PATTERN = re.compile(r"^(\d+[smh])+$", re.IGNORECASE)
+_DURATION_COMPONENT = re.compile(r"(\d+)([smh])", re.IGNORECASE)
+
+
+class DurationParseError(ValueError):
+    """Raised when a duration string cannot be parsed."""
+
+    pass
+
+
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/magpie/cli/config.py]: Fixed in commit 2334ff9. Removed the unnecessary `pass` statement from the `DurationParseError` class.

File: src/magpie/cli/config.py [hunk 2]
```diff
@@ -121,6 +122,67 @@ def get_token(
...
+    """
+    value = value.strip()
+    if not value:
+        raise DurationParseError("Empty duration string")
+
+    # Try plain number first (seconds)
+    try:
+        return int(float(value))
+    except ValueError:
+        pass
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/magpie/cli/config.py]: Fixed in commit 2334ff9. Added validation to reject negative durations - `parse_duration()` now raises `DurationParseError` with the message "Negative durations are not allowed" for negative values.

File: tests/unit/test_cli_config.py [hunk 3]
```diff
@@ -256,9 +258,105 @@ def test_cli_override_zero_is_valid(self, monkeypatch: pytest.MonkeyPatch) -> No
...
+        with pytest.raises(DurationParseError, match="Invalid duration format"):
+            parse_duration("1d")  # days not supported
+
+    def test_mixed_invalid_raises(self) -> None:
+        """Test that mixed valid/invalid formats raise DurationParseError."""
+        with pytest.raises(DurationParseError, match="Invalid duration format"):
+            parse_duration("1h30x")
+
+        with pytest.raises(DurationParseError, match="Invalid duration format"):
+            parse_duration("5m abc")
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/unit/test_cli_config.py:362]: Fixed in commit 2334ff9. Added two new test methods:
- `test_negative_plain_seconds_raises`: Tests that `parse_duration("-30")` and `parse_duration("-100")` raise `DurationParseError`
- `test_negative_float_seconds_raises`: Tests that `parse_duration("-45.5")` raises `DurationParseError`
